src/main/kotlin/sidechain/eth/consumer/EthConsumer.kt

```diff
@@ -1,13 +1,48 @@
 package sidechain.eth.consumer
 
+import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.success
 import mu.KLogging
+import org.web3j.utils.Numeric
+import sidechain.eth.util.DeployHelper
 import withdrawalservice.WithdrawalServiceOutputEvent
 
 class EthConsumer {
+    private val deployHelper = DeployHelper()
 
-    // TODO a.chernyshov: rework with effective implementation
-    fun consume(event: WithdrawalServiceOutputEvent) {
-        logger.info { "consumed eth event" }
+    fun consume(event: Result<WithdrawalServiceOutputEvent, Exception>) {
```

Comment:
I think it should accept `WithdrawalServiceOutputEvent`, not result. See Railway-Oriented approach (https://github.com/swlaschin/Railway-Oriented-Programming-Example)

Reply:
And success-failure branching should be processed in monad style.